beta-operator-rewards/rewards-requirements.ts

```diff
@@ -0,0 +1,229 @@
+  const preParamsAvgInterval = 24 // hours
+  const preParamsResolution = 24 // hours
+  const minPreParams = 500 // min requirement for pre params daily avg
+  const factors = {
+    isBeaconAuthorized: "isBeaconAuthorized",
+    isTbtcAuthorized: "isTbtcAuthorized",
+    upTime: "upTime",
+    version: "verion",
+    preParams: "preParams",
+  };
+  const upTimeRewardsCoefficient = "upTimeRewardsCoefficient"
+  const peersDataFile = options.output
+  const defaultProvider = "goerli";
+
+  // Query for bootstrap data that has peer instances
+  const queryBootstrapData = `${prometheusAPI}/query_range`;
+  const paramsBootstrapData = {
+    query: `up{job='${prometheous_job}'}`,
+    start: startRewardsTimestamp,
+    end: endRewardsTimestamp,
+    step: queryStep,
+  };
+
+  const bootstrapData = await queryPrometheus(queryBootstrapData, paramsBootstrapData);
+
+  let peersData = new Map<string, Map<string, number>>(); // peer address -> {component name: factor}
+
+  const provider = ethers.getDefaultProvider(defaultProvider);
+  const randomBeacon = new Contract(
+    RandomBeaconAddress,
+    JSON.stringify(RandomBeaconABI),
+    provider
+  );
+
+  const walletRegistry = new Contract(
+    WalletRegistryAddress,
+    JSON.stringify(WalletRegistryABI),
+    provider
+  );
+
+  if ((Date.now() / 1000) < endRewardsTimestamp) {
+    return "End time interval must be in the past";
+  }
+
+  let counter = 0
+  for (let i = 0; i < await bootstrapData.data.result.length; i++) {
+    const peer = bootstrapData.data.result[i]
+    let peerData = new Map<string, any>(); // Map<factor_name: value> value is in the range from 0 to 1
+    peerData.set("address", peer.metric.chain_address)
+
+    /// Random Beacon application authorization requirement
+
+    const stakingProviderAddressForBeacon =
+      await randomBeacon.operatorToStakingProvider(peer.metric.chain_address);
+    const eligibleStakeForBeacon = await walletRegistry.eligibleStake(
+      stakingProviderAddressForBeacon
+    );
+    if (eligibleStakeForBeacon.isZero()) {
+      peerData.set(factors.isBeaconAuthorized, 0);
+    } else {
+      peerData.set(factors.isBeaconAuthorized, 1);
+    }
+
+    /// tBTC application authorized requirement
+
+    const stakingProviderAddressForTbtc =
+      await walletRegistry.operatorToStakingProvider(peer.metric.chain_address);
+    const eligibleStakeForTbtc = await walletRegistry.eligibleStake(
+      stakingProviderAddressForTbtc
+    );
+    if (eligibleStakeForTbtc.isZero()) {
+      peerData.set(factors.isTbtcAuthorized, 0);
+    } else {
+      peerData.set(factors.isTbtcAuthorized, 1);
+    }
+
+    /// Up time requirement
+
+    // First registered 'up' metric in a given timeframe <start:end>. We start
+    // evaluating uptime from this point.
+    const firstRegisteredUptime = peer.values[0][0];
+    const uptimeSearchRange = endRewardsTimestamp - firstRegisteredUptime;
+    const queryUptime = `${prometheusAPI}/query`;
```

Comment:
multiple variables are bound to this same string: `${prometheusAPI}/query`

I think we can rename it something generic and only bind it once

Reply:
Sure, as part of [982a1f0](https://github.com/keep-network/keep-core/pull/3346/commits/982a1f01cbd3036fa4c035d697edb323b75493c6)